src/backend/expungeservice/expunger/expunger.py

```diff
@@ -92,7 +94,7 @@ def _set_most_recent_convictions(self):
             self.second_most_recent_conviction = self.convictions[-2]
 
     def _assign_most_recent_charge(self):
-        self.charges.sort(key=lambda charge: charge.disposition.date, reverse=True)
+        self.charges.sort(key=lambda charge: charge.disposition.date if charge.disposition.date else (date_class.today() + relativedelta(years=-1000)), reverse=True)
```

Comment:
Question re Expunger intended behavior: Does a charge belong in this list if there is no disposition? If there is no disposition then the charge has no date information and can't be sorted by this. It sounlds like you're suggesting that a charge without a ruling doesn't belong in this list. Is that correct?

The change here is meant to cover the edge case in which at least one charge has a disposition and at least one does not. This is exclusive with the edge case that this list has only one charge.  If a Record has only one Charge and it has no Disposition, is it valid to call it the most recent charge? If yes, then this implementation is valid.

Reply:
@wittejm @aemrich Yes, these methods are doing to much and are checking for null.... Thanks for pointing that out. We can still DRY them out a bit by doing this:
```
    def acquitted(self):
        return self.disposition.ruling and self.disposition.ruling[0:9].lower() != 'convicted'

    def convicted(self):
        return self.disposition.ruling and not self.acquitted()
```
However both the methods are still checking for null and that could be resolved by implementing the null object pattern here. 

I'm fine with the current code in its current state, however upon merging I'd like to create a refactoring ticket to dry the code out and remove the check for null.

Reply:
Ok great, thanks. @NickSchimek I've made your changes to the acquitted and convicted methods -- I'll go ahead and merge this in now.

Reply:
Thanks for your work on this @aemrich 
In regards to @wittejm questions:
> Does a charge belong in this list if there is no disposition?

No a charge without a disposition should not be in this list. The most recent charge is used in the Time Analyzer to determine the time eligibility for some class b felonies. If a charge without a disposition is put into this list and falls through to the time analyzer then the error is repeated there.

> If there is no disposition then the charge has no date information and can't be sorted by this. It sounds like you're suggesting that a charge without a ruling doesn't belong in this list. Is that correct?

Yes, a charge without a disposition, hence no ruling/date, should not be in this list; because it cannot be evaluated correctly and should be flagged so that we can bring it to the user's attention.

> The change here is meant to cover the edge case in which at least one charge has a disposition and at least one does not.

Not necessarily. That was the case issue provided to expose the bug, hence the reason I also requested unit tests to be added in the original issue. And the case detail `CASE_WITHOUT_DISPOS` can be used to expose the bug too. 

> If a Record has only one Charge and it has no Disposition, is it valid to call it the most recent charge?

No, it should be labeled as the most recent charge, because we could not run time analysis using a charge without a disposition date. Therefore we need to flag the charge and mark it as not having a disposition (as mentioned in the issue) so that the user can be notified to evaluate independently.